File: scenarios/basic/mobile/react-native/n5.js
import React, { useState } from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
  Image,
  Linking,
} from 'react-native';

const GenericLinkText = () => {
  const [selectedArticle, setSelectedArticle] = useState(null);

  const articles = [
    {
      id: 1,
      title: 'Understanding React Native Performance Optimization',
      excerpt: 'Learn how to optimize your React Native applications for better performance and user experience.',
      category: 'Development',
      readTime: '5 min read',
      author: 'John Smith',
      date: '2024-01-15',
      url: 'https://example.com/react-native-performance'
    },
    {
      id: 2,
      title: 'Mobile UI/UX Design Best Practices',
      excerpt: 'Essential design principles for creating intuitive and accessible mobile interfaces.',
      category: 'Design',
      readTime: '7 min read',
      author: 'Sarah Johnson',
      date: '2024-01-12',
      url: 'https://example.com/mobile-ui-design'
    },
    {
      id: 3,
      title: 'Cross-Platform Development Strategies',
      excerpt: 'Compare different approaches to building apps that work across multiple platforms.',
      category: 'Development',
      readTime: '10 min read',
      author: 'Mike Chen',
      date: '2024-01-10',
      url: 'https://example.com/cross-platform-strategies'
    },
    {
      id: 4,
      title: 'State Management in Modern React',
      excerpt: 'Explore different state management solutions for complex React applications.',
      category: 'Development',
      readTime: '8 min read',
      author: 'Emily Davis',
      date: '2024-01-08',
      url: 'https://example.com/react-state-management'
    }
  ];

  const handleReadMore = (url) => {
    Linking.openURL(url);
  };

  const handleArticlePress = (article) => {
    setSelectedArticle(article);
  };

  const formatDate = (dateString) => {
    const date = new Date(dateString);
    return date.toLocaleDateString('en-US', {
      year: 'numeric',
      month: 'long',
      day: 'numeric'
    });
  };

  return (
    <ScrollView style={styles.container}>
      {/* Header */}
      <View style={styles.header}>
        <Text style={styles.headerTitle}>Tech Blog</Text>
        <Text style={styles.headerSubtitle}>Latest articles and insights</Text>
      </View>

      {/* Featured Article */}
      <View style={styles.featuredContainer}>
        <Text style={styles.sectionTitle}>Featured Article</Text>
        <View style={styles.featuredCard}>
          <Image
            source={{ uri: 'https://via.placeholder.com/400x200' }}
            style={styles.featuredImage}
            accessibilityLabel="Featured article image"
          />
          <View style={styles.featuredContent}>
            <Text style={styles.featuredCategory}>Development</Text>
            <Text style={styles.featuredTitle}>
              Building Scalable React Native Applications
            </Text>
            <Text style={styles.featuredExcerpt}>
              Discover the key principles and patterns for creating maintainable and scalable React Native applications that can grow with your business needs.
            </Text>
            <View style={styles.featuredMeta}>
              <Text style={styles.featuredAuthor}>Alex Thompson</Text>
              <Text style={styles.featuredDate}>January 20, 2024</Text>
              <Text style={styles.featuredReadTime}>12 min read</Text>
            </View>
            <TouchableOpacity 
              style={styles.featuredButton}
              onPress={() => handleReadMore('https://example.com/scalable-react-native')}
            >
              <Text style={styles.featuredButtonText}>Read more</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>

      {/* Articles List */}
      <View style={styles.articlesContainer}>
        <Text style={styles.sectionTitle}>Recent Articles</Text>
        {articles.map((article) => (
          <View key={article.id} style={styles.articleCard}>
            <View style={styles.articleImageContainer}>
              <Image
                source={{ uri: 'https://via.placeholder.com/120x80' }}
                style={styles.articleImage}
                accessibilityLabel={`Article image for ${article.title}`}
              />
            </View>
            
            <View style={styles.articleContent}>
              <View style={styles.articleHeader}>
                <Text style={styles.articleCategory}>{article.category}</Text>
                <Text style={styles.articleReadTime}>{article.readTime}</Text>
              </View>
              
              <Text style={styles.articleTitle}>{article.title}</Text>
              <Text style={styles.articleExcerpt}>{article.excerpt}</Text>
              
              <View style={styles.articleMeta}>
                <Text style={styles.articleAuthor}>By {article.author}</Text>
                <Text style={styles.articleDate}>{formatDate(article.date)}</Text>
              </View>
              
              <View style={styles.articleActions}>
                <TouchableOpacity 
                  style={styles.readMoreButton}
                  onPress={() => handleReadMore(article.url)}
                >
                  <Text style={styles.readMoreText}>Read more</Text>
                </TouchableOpacity>
                
                <TouchableOpacity 
                  style={styles.shareButton}
                  onPress={() => handleArticlePress(article)}
                >
                  <Text style={styles.shareText}>Share</Text>
                </TouchableOpacity>
              </View>
            </View>
          </View>
        ))}
      </View>

      {/* Newsletter Signup */}
      <View style={styles.newsletterContainer}>
        <Text style={styles.newsletterTitle}>Stay Updated</Text>
        <Text style={styles.newsletterDescription}>
          Get the latest articles delivered to your inbox
        </Text>
        <TouchableOpacity style={styles.newsletterButton}>
          <Text style={styles.newsletterButtonText}>Subscribe</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.newsletterLink}>
          <Text style={styles.newsletterLinkText}>Read more</Text>
        </TouchableOpacity>
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F8F9FA',
  },
  header: {
    backgroundColor: '#2C3E50',
    padding: 24,
    alignItems: 'center',
  },
  headerTitle: {
    fontSize: 28,
    fontWeight: 'bold',
    color: '#FFFFFF',
    marginBottom: 8,
  },
  headerSubtitle: {
    fontSize: 16,
    color: '#BDC3C7',
  },
  featuredContainer: {
    padding: 16,
  },
  sectionTitle: {
    fontSize: 20,
    fontWeight: '600',
    color: '#2C3E50',
    marginBottom: 16,
  },
  featuredCard: {
    backgroundColor: '#FFFFFF',
    borderRadius: 12,
    overflow: 'hidden',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 4,
  },
  featuredImage: {
    width: '100%',
    height: 200,
  },
  featuredContent: {
    padding: 20,
  },
  featuredCategory: {
    fontSize: 14,
    color: '#3498DB',
    fontWeight: '600',
    marginBottom: 8,
  },
  featuredTitle: {
    fontSize: 22,
    fontWeight: 'bold',
    color: '#2C3E50',
    marginBottom: 12,
    lineHeight: 28,
  },
  featuredExcerpt: {
    fontSize: 16,
    color: '#7F8C8D',
    lineHeight: 24,
    marginBottom: 16,
  },
  featuredMeta: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 20,
  },
  featuredAuthor: {
    fontSize: 14,
    color: '#7F8C8D',
  },
  featuredDate: {
    fontSize: 14,
    color: '#7F8C8D',
  },
  featuredReadTime: {
    fontSize: 14,
    color: '#7F8C8D',
  },
  featuredButton: {
    backgroundColor: '#3498DB',
    paddingVertical: 12,
    paddingHorizontal: 24,
    borderRadius: 8,
    alignSelf: 'flex-start',
  },
  featuredButtonText: {
    color: '#FFFFFF',
    fontWeight: '600',
    fontSize: 16,
  },
  articlesContainer: {
    padding: 16,
  },
  articleCard: {
    backgroundColor: '#FFFFFF',
    borderRadius: 12,
    padding: 16,
    marginBottom: 16,
    flexDirection: 'row',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 2,
  },
  articleImageContainer: {
    width: 120,
    height: 80,
    marginRight: 16,
  },
  articleImage: {
    width: 120,
    height: 80,
    borderRadius: 8,
  },
  articleContent: {
    flex: 1,
  },
  articleHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 8,
  },
  articleCategory: {
    fontSize: 12,
    color: '#3498DB',
    fontWeight: '600',
  },
  articleReadTime: {
    fontSize: 12,
    color: '#7F8C8D',
  },
  articleTitle: {
    fontSize: 16,
    fontWeight: '600',
    color: '#2C3E50',
    marginBottom: 8,
    lineHeight: 22,
  },
  articleExcerpt: {
    fontSize: 14,
    color: '#7F8C8D',
    lineHeight: 20,
    marginBottom: 12,
  },
  articleMeta: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 12,
  },
  articleAuthor: {
    fontSize: 12,
    color: '#7F8C8D',
  },
  articleDate: {
    fontSize: 12,
    color: '#7F8C8D',
  },
  articleActions: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  readMoreButton: {
    backgroundColor: '#3498DB',
    paddingVertical: 8,
    paddingHorizontal: 16,
    borderRadius: 6,
  },
  readMoreText: {
    color: '#FFFFFF',
    fontWeight: '600',
    fontSize: 14,
  },
  shareButton: {
    backgroundColor: '#FFFFFF',
    borderWidth: 1,
    borderColor: '#3498DB',
    paddingVertical: 8,
    paddingHorizontal: 16,
    borderRadius: 6,
  },
  shareText: {
    color: '#3498DB',
    fontWeight: '600',
    fontSize: 14,
  },
  newsletterContainer: {
    backgroundColor: '#34495E',
    margin: 16,
    padding: 24,
    borderRadius: 12,
    alignItems: 'center',
  },
  newsletterTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#FFFFFF',
    marginBottom: 8,
  },
  newsletterDescription: {
    fontSize: 16,
    color: '#BDC3C7',
    textAlign: 'center',
    marginBottom: 20,
  },
  newsletterButton: {
    backgroundColor: '#E74C3C',
    paddingVertical: 12,
    paddingHorizontal: 32,
    borderRadius: 8,
    marginBottom: 12,
  },
  newsletterButtonText: {
    color: '#FFFFFF',
    fontWeight: '600',
    fontSize: 16,
  },
  newsletterLink: {
    paddingVertical: 8,
  },
  newsletterLinkText: {
    color: '#3498DB',
    fontSize: 14,
  },
});

export default GenericLinkText;

File: scenarios/basic/mobile/flutter/n5.dart
import 'package:flutter/material.dart';

class GenericLinkText extends StatefulWidget {
  const GenericLinkText({Key? key}) : super(key: key);

  @override
  State<GenericLinkText> createState() => _GenericLinkTextState();
}

class _GenericLinkTextState extends State<GenericLinkText> {
  int selectedCategory = 0;
  final List<String> categories = ['All', 'Technology', 'Design', 'Business', 'Science'];

  final List<Map<String, dynamic>> articles = [
    {
      'id': 1,
      'title': 'Understanding Flutter Performance Optimization',
      'excerpt': 'Learn how to optimize your Flutter applications for better performance and user experience.',
      'category': 'Technology',
      'readTime': '5 min read',
      'author': 'John Smith',
      'date': '2024-01-15',
      'image': 'https://via.placeholder.com/300x200',
      'url': 'https://example.com/flutter-performance',
    },
    {
      'id': 2,
      'title': 'Mobile UI/UX Design Best Practices',
      'excerpt': 'Essential design principles for creating intuitive and accessible mobile interfaces.',
      'category': 'Design',
      'readTime': '7 min read',
      'author': 'Sarah Johnson',
      'date': '2024-01-12',
      'image': 'https://via.placeholder.com/300x200',
      'url': 'https://example.com/mobile-ui-design',
    },
    {
      'id': 3,
      'title': 'Cross-Platform Development Strategies',
      'excerpt': 'Compare different approaches to building apps that work across multiple platforms.',
      'category': 'Technology',
      'readTime': '10 min read',
      'author': 'Mike Chen',
      'date': '2024-01-10',
      'image': 'https://via.placeholder.com/300x200',
      'url': 'https://example.com/cross-platform-strategies',
    },
    {
      'id': 4,
      'title': 'State Management in Modern React',
      'excerpt': 'Explore different state management solutions for complex React applications.',
      'category': 'Technology',
      'readTime': '8 min read',
      'author': 'Emily Davis',
      'date': '2024-01-08',
      'image': 'https://via.placeholder.com/300x200',
      'url': 'https://example.com/react-state-management',
    },
    {
      'id': 5,
      'title': 'Building Scalable React Native Applications',
      'excerpt': 'Discover the key principles and patterns for creating maintainable and scalable React Native applications.',
      'category': 'Technology',
      'readTime': '12 min read',
      'author': 'Alex Thompson',
      'date': '2024-01-20',
      'image': 'https://via.placeholder.com/300x200',
      'url': 'https://example.com/scalable-react-native',
    },
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.grey[50],
      appBar: AppBar(
        title: const Text('Tech Blog'),
        backgroundColor: Colors.blue[700],
        foregroundColor: Colors.white,
        elevation: 0,
        actions: [
          IconButton(
            icon: const Icon(Icons.search),
            onPressed: () {},
          ),
          IconButton(
            icon: const Icon(Icons.notifications),
            onPressed: () {},
          ),
        ],
      ),
      body: Column(
        children: [
          // Featured Article
          _buildFeaturedArticle(),
          
          // Category Filter
          Container(
            height: 50,
            padding: const EdgeInsets.symmetric(vertical: 8),
            child: ListView.builder(
              scrollDirection: Axis.horizontal,
              padding: const EdgeInsets.symmetric(horizontal: 16),
              itemCount: categories.length,
              itemBuilder: (context, index) {
                final isSelected = selectedCategory == index;
                return Padding(
                  padding: const EdgeInsets.only(right: 12),
                  child: FilterChip(
                    label: Text(categories[index]),
                    selected: isSelected,
                    onSelected: (selected) {
                      setState(() {
                        selectedCategory = index;
                      });
                    },
                    backgroundColor: Colors.white,
                    selectedColor: Colors.blue[100],
                    checkmarkColor: Colors.blue[700],
                  ),
                );
              },
            ),
          ),
          
          // Articles List
          Expanded(
            child: ListView.builder(
              padding: const EdgeInsets.all(16),
              itemCount: articles.length,
              itemBuilder: (context, index) {
                return _buildArticleCard(articles[index]);
              },
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildFeaturedArticle() {
    return Container(
      margin: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(12),
        boxShadow: [
          BoxShadow(
            color: Colors.grey.withOpacity(0.1),
            spreadRadius: 1,
            blurRadius: 4,
            offset: const Offset(0, 2),
          ),
        ],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          // Featured Image
          Container(
            width: double.infinity,
            height: 200,
            decoration: const BoxDecoration(
              borderRadius: BorderRadius.only(
                topLeft: Radius.circular(12),
                topRight: Radius.circular(12),
              ),
              image: DecorationImage(
                image: NetworkImage('https://via.placeholder.com/400x200'),
                fit: BoxFit.cover,
              ),
            ),
          ),
          
          // Featured Content
          Padding(
            padding: const EdgeInsets.all(20),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Container(
                  padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                  decoration: BoxDecoration(
                    color: Colors.blue[100],
                    borderRadius: BorderRadius.circular(12),
                  ),
                  child: const Text(
                    'Featured',
                    style: TextStyle(
                      fontSize: 12,
                      fontWeight: FontWeight.bold,
                      color: Colors.blue,
                    ),
                  ),
                ),
                
                const SizedBox(height: 12),
                
                const Text(
                  'Building Scalable React Native Applications',
                  style: TextStyle(
                    fontSize: 22,
                    fontWeight: FontWeight.bold,
                    color: Colors.black87,
                  ),
                ),
                
                const SizedBox(height: 8),
                
                const Text(
                  'Discover the key principles and patterns for creating maintainable and scalable React Native applications that can grow with your business needs.',
                  style: TextStyle(
                    fontSize: 16,
                    color: Colors.grey,
                    height: 1.5,
                  ),
                ),
                
                const SizedBox(height: 16),
                
                Row(
                  children: [
                    const CircleAvatar(
                      radius: 16,
                      backgroundColor: Colors.blue,
                      child: Text(
                        'AT',
                        style: TextStyle(
                          fontSize: 12,
                          fontWeight: FontWeight.bold,
                          color: Colors.white,
                        ),
                      ),
                    ),
                    const SizedBox(width: 8),
                    const Text(
                      'Alex Thompson',
                      style: TextStyle(
                        fontSize: 14,
                        fontWeight: FontWeight.w500,
                        color: Colors.black87,
                      ),
                    ),
                    const SizedBox(width: 16),
                    Text(
                      'January 20, 2024',
                      style: TextStyle(
                        fontSize: 12,
                        color: Colors.grey[600],
                      ),
                    ),
                    const SizedBox(width: 16),
                    Text(
                      '12 min read',
                      style: TextStyle(
                        fontSize: 12,
                        color: Colors.grey[600],
                      ),
                    ),
                  ],
                ),
                
                const SizedBox(height: 20),
                
                // GENERIC LINK TEXT - "Read more" without context
                GestureDetector(
                  onTap: () {
                    // Navigate to full article
                  },
                  child: Container(
                    padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 12),
                    decoration: BoxDecoration(
                      color: Colors.blue[700],
                      borderRadius: BorderRadius.circular(8),
                    ),
                    child: const Text(
                      'Read more', // GENERIC LINK TEXT - No context about destination
                      style: TextStyle(
                        color: Colors.white,
                        fontSize: 16,
                        fontWeight: FontWeight.w600,
                      ),
                    ),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildArticleCard(Map<String, dynamic> article) {
    return Container(
      margin: const EdgeInsets.only(bottom: 16),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(12),
        boxShadow: [
          BoxShadow(
            color: Colors.grey.withOpacity(0.1),
            spreadRadius: 1,
            blurRadius: 4,
            offset: const Offset(0, 2),
          ),
        ],
      ),
      child: Row(
        children: [
          // Article Image
          Container(
            width: 120,
            height: 100,
            decoration: const BoxDecoration(
              borderRadius: BorderRadius.only(
                topLeft: Radius.circular(12),
                bottomLeft: Radius.circular(12),
              ),
              image: DecorationImage(
                image: NetworkImage('https://via.placeholder.com/120x100'),
                fit: BoxFit.cover,
              ),
            ),
          ),
          
          // Article Content
          Expanded(
            child: Padding(
              padding: const EdgeInsets.all(16),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  // Category and Read Time
                  Row(
                    children: [
                      Container(
                        padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 2),
                        decoration: BoxDecoration(
                          color: Colors.blue[100],
                          borderRadius: BorderRadius.circular(4),
                        ),
                        child: Text(
                          article['category'],
                          style: TextStyle(
                            fontSize: 10,
                            fontWeight: FontWeight.bold,
                            color: Colors.blue[700],
                          ),
                        ),
                      ),
                      const SizedBox(width: 8),
                      Text(
                        article['readTime'],
                        style: TextStyle(
                          fontSize: 10,
                          color: Colors.grey[600],
                        ),
                      ),
                    ],
                  ),
                  
                  const SizedBox(height: 8),
                  
                  // Article Title
                  Text(
                    article['title'],
                    style: const TextStyle(
                      fontSize: 16,
                      fontWeight: FontWeight.bold,
                      color: Colors.black87,
                    ),
                    maxLines: 2,
                    overflow: TextOverflow.ellipsis,
                  ),
                  
                  const SizedBox(height: 4),
                  
                  // Article Excerpt
                  Text(
                    article['excerpt'],
                    style: TextStyle(
                      fontSize: 12,
                      color: Colors.grey[600],
                      height: 1.4,
                    ),
                    maxLines: 2,
                    overflow: TextOverflow.ellipsis,
                  ),
                  
                  const SizedBox(height: 8),
                  
                  // Author and Date
                  Row(
                    children: [
                      Text(
                        'By ${article['author']}',
                        style: TextStyle(
                          fontSize: 10,
                          color: Colors.grey[600],
                        ),
                      ),
                      const SizedBox(width: 8),
                      Text(
                        '•',
                        style: TextStyle(
                          fontSize: 10,
                          color: Colors.grey[600],
                        ),
                      ),
                      const SizedBox(width: 8),
                      Text(
                        article['date'],
                        style: TextStyle(
                          fontSize: 10,
                          color: Colors.grey[600],
                        ),
                      ),
                    ],
                  ),
                  
                  const SizedBox(height: 12),
                  
                  // Action Buttons
                  Row(
                    children: [
                      // GENERIC LINK TEXT - "Read more" without context
                      GestureDetector(
                        onTap: () {
                          // Navigate to full article
                        },
                        child: Container(
                          padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
                          decoration: BoxDecoration(
                            color: Colors.blue[700],
                            borderRadius: BorderRadius.circular(6),
                          ),
                          child: const Text(
                            'Read more', // GENERIC LINK TEXT - No context about destination
                            style: TextStyle(
                              color: Colors.white,
                              fontSize: 12,
                              fontWeight: FontWeight.w600,
                            ),
                          ),
                        ),
                      ),
                      
                      const SizedBox(width: 8),
                      
                      // Share Button
                      GestureDetector(
                        onTap: () {
                          // Share functionality
                        },
                        child: Container(
                          padding: const EdgeInsets.all(6),
                          decoration: BoxDecoration(
                            color: Colors.grey[200],
                            borderRadius: BorderRadius.circular(6),
                          ),
                          child: const Icon(
                            Icons.share,
                            size: 16,
                            color: Colors.grey,
                          ),
                        ),
                      ),
                    ],
                  ),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }
}
